modules/login/src/main/resources/META-INF/resources/login/js/registration.js

```diff
@@ -0,0 +1,401 @@
+
+                var sortableOrgTypes = [];
+                //Push the object into an array for sorting.
+                for (var orgType in orgLookupData.organizationTypes) {
+                    sortableOrgTypes.push([orgType, orgLookupData.organizationTypes[orgType]]);
+                }
+
+                //sort by name before populating the select options.
+                sortableOrgTypes.sort(sortobjArrayByName);
+                //populate the options now that they are sorted properly.
+                populateSelect($($selectOrgType),sortableOrgTypes);
+                $($selectOrgType).selectpicker('refresh'); //Refresh because we just added a bunch of items
+
+                //Extract the organizations of four specific OrgTypes: 'USAR', 'Federal IMT', 'CDF IMT', and 'Other Local IMT'
+                // Put those organizations as options, each having its own dropdown list.
+                populateSelect($($selectImtUSAR),getOrgsData(orgLookupData,"USAR").sort(sortobjArrayByName));
+                $($selectImtUSAR).selectpicker('refresh');
+                populateSelect($($selectImtFed),getOrgsData(orgLookupData,"Federal IMT").sort(sortobjArrayByName));
+                $($selectImtFed).selectpicker('refresh');
+                populateSelect($($selectImtCDF),getOrgsData(orgLookupData,"CDF IMT").sort(sortobjArrayByName));
+                $($selectImtCDF).selectpicker('refresh');
+                populateSelect($($selectImtLocal),getOrgsData(orgLookupData,"Other Local IMT").sort(sortobjArrayByName));
+                $($selectImtLocal).selectpicker('refresh');
+                return true;
+            })
+            .fail(function(result, errortext, error) {
+                console.log(errortext);
+                console.log(error);
+                $($dlgStep1).addClass('hidden');
+                doError('Failed to get Organization Data from server.');
+                return false;
+            });
+
+        //Create a lookup map for easy filtering.
+        // Takes in an array of objects.
+        // Each object must have an id property.
+        // return a lookup object with the ids as properties (keys) to the object.
+        function createLookup(arrIdTypes) {
+            var lookup = {};
+
+            //turn the id into a property to make this into a map(idTypeid,idType).
+            arrIdTypes.forEach(function(idType) {
+                lookup[idType.id] = idType;
+            });
+            return lookup;
+        }
+    }
+
+    function onOrgAffilliatesChanged() {
+        $('#selectOrganizations').removeAttr('disabled');
+        $('#btnNext1').attr('disabled', 'disabled');
+        filterOrgs($('#selectOrgAffiliates').val());
+
+    }
+
+    function onOrgChanged() {
+        console.log($('#selectOrganizations').val());
+        $('#btnNext1').prop('disabled', false);
+    }
+    function filterOrgs(OrgAffiliateId) {
+        // console.log('Org Affilated Id: ' + OrgAffiliateId);
```

Comment:
Keep in mind that commented out code such as this, doesn't make it to the customer. The minified code strips out all comments. I leave this one in for debugging.